ambari-server/src/main/java/org/apache/ambari/server/controller/internal/HostStackVersionResourceProvider.java

```diff
@@ -220,17 +221,14 @@ public HostStackVersionResourceProvider(
    * @param clusterName name of cluster or null if no any
    */
   public void addRequestedEntities(Set<Resource> resources,
-                                   List<HostVersionEntity> requestedEntities,
+      List<MpackHostStateEntity> requestedEntities,
                                    Set<String> requestedIds,
                                    String clusterName) {
-    for (HostVersionEntity entity: requestedEntities) {
-      StackId stackId = new StackId(entity.getRepositoryVersion().getStack());
-
-      RepositoryVersionEntity repoVerEntity = repositoryVersionDAO.findByStackAndVersion(
-          stackId, entity.getRepositoryVersion().getVersion());
-
+    for (MpackHostStateEntity entity : requestedEntities) {
       final Resource resource = new ResourceImpl(Resource.Type.HostStackVersion);
 
+      MpackEntity mpackEntity = entity.getMpack();
+      StackId stackId = new StackId(mpackEntity.getMpackName(), mpackEntity.getMpackVersion());
```

Comment:
Just a minor nit: I would imagine this is something that would happen a lot, so either have mpack.getStackId() or new StackId(mpackEntity) or something.

Reply:
Sure - I can add that.